libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/MetadataFuzzer.java

```diff
@@ -140,45 +148,142 @@ private void initializeFuzzer(String serviceName, ModelObject objectToBeFuzzed,
+  }
+
+  private boolean isObjectProperty(ModelPropertyType type) {
+    if (type == ModelPropertyType.OBJECT || type == ModelPropertyType.ARRAY
+        || type == ModelPropertyType.REF_SCHEMA || type == ModelPropertyType.REF_DEFINITION) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
+
   /**
    * Method which turns the properties of the model into a dataprovider for the fuzzer to use.
    * 
    * @param set JSONObject representing a set of metadata nodes
    * @param fuzzSet the array in which the data will be pushed
-   * @param propertyType indicator specifying links or root property
+   * @param fuzzPropertyType indicator specifying links or root property
    * @throws UnexpectedModelPropertyTypeException
    */
-  private void populateFuzzSet(JSONObject set, Object[][] fuzzSet, String propertyType) {
+  private Object[][] populateFuzzSet(ModelObject objectToBeFuzzed,
+      ArrayList<Object[]> fuzzPropertiesList, String currentPath, JSONObject currentPathModel) {
+    Object[][] returnArray = null;
     try {
-      Object[] keys = set.keySet().toArray();
+      ModelProperties modelProps = getObjectProperties(objectToBeFuzzed, currentPathModel);
+      Object[] propertiesArray = modelProps.propertiesArray;
+      JSONObject properties = modelProps.properties;
+      for (int i = 0; i < propertiesArray.length; i++) {
+        String propertyName = propertiesArray[i].toString();
+        
+        JSONObject propertyDef = (JSONObject) properties.get(propertyName);
+        ModelPropertyType type = ModelObject.getModelPropertyType(propertyDef);
+        if (isObjectProperty(type)) {
+          populateFuzzSet(objectToBeFuzzed,
+              fuzzPropertiesList, getFuzzPath(propertyDef, currentPath, propertyName),
+              propertyDef);
+        } else {
+          Object[] propertyFuzzSet = new Object[3];
+          propertyFuzzSet[0] = currentPath;
+          propertyFuzzSet[1] = propertyName;
+          propertyFuzzSet[2] = (ModelPropertyType) type;
 
-      for (int i = 0; i < keys.length; i++) {
-        JSONObject obj = (JSONObject) set.get(keys[i]);
-        ModelPropertyType type = ModelObject.getModelPropertyType(obj);
+          fuzzPropertiesList.add(propertyFuzzSet);
+        }
+      }
 
-        fuzzSet[i][0] = keys[i].toString();
-        fuzzSet[i][1] = (ModelPropertyType) type;
-        fuzzSet[i][2] = propertyType;
+      if (currentPath == root) {
+        returnArray = convertFuzzListToArray(fuzzPropertiesList);
       }
     } catch (UnexpectedModelPropertyTypeException e) {
       System.out.println("FUZZER ERROR: UnexpectedModelPropertyType:" + e.getMessage());
       e.printStackTrace();
     }
+    return returnArray;
+  }
+
+  private Object[][] convertFuzzListToArray(ArrayList<Object[]> list) {
+    int len = list.size();
+    Object[][] returnArray = new Object[len][3];
+    for (int i = 0; i < len; i++) {
+      returnArray[i] = list.get(i);
+    }
 
+    return returnArray;
```

Comment:
TestNG requires a native array for dataproviders, so need to convert the ArrayList to something useable there.